tests/sentry/lang/javascript/test_processing.py

```diff
@@ -120,3 +121,197 @@ def test_node_modules_regex(self):
+                        },
+                        {
+                            "abs_path": "webpack:///node_modules/react/index.js",
+                            "filename": "./node_modules/react/index.js",  # Note ./ prefix
+                            "lineno": 20,
+                            "colno": 30,
+                            "function": "createElement",
+                            "data": {
+                                "symbolicated": True,
+                                "sourcemap": "webpack:///node_modules/react/index.js.map",
+                                "resolved_with": "source-map",
+                            },
+                            "platform": "javascript",
+                        },
+                    ]
+                }
+            ],
+            "raw_stacktraces": [
+                {
+                    "frames": [
+                        {
+                            "abs_path": "webpack:///app/components/App.jsx",
+                            "filename": "app/components/App.jsx",
+                            "lineno": 10,
+                            "colno": 15,
+                            "function": "render",
+                            "platform": "javascript",
+                        },
+                        {
+                            "abs_path": "webpack:///node_modules/react/index.js",
+                            "filename": "node_modules/react/index.js",
+                            "lineno": 20,
+                            "colno": 30,
+                            "function": "createElement",
+                            "platform": "javascript",
+                        },
+                    ]
+                }
+            ],
+        }
+        return data, symbolicator
+
+    def test_process_js_stacktraces_with_symbolicated_in_app_frames(self):
+        """Test symbolicated_in_app is True when all in-app frames are symbolicated"""
+        data, symbolicator = self._get_test_data_and_symbolicator()
+
+        result = process_js_stacktraces(symbolicator, data)
+        self.assertTrue(result["symbolicated_in_app"])
+
+    def test_process_js_stacktraces_with_unsymbolicated_in_app_frames(self):
+        """Test symbolicated_in_app is False when in-app frames are not symbolicated"""
+        data, symbolicator = self._get_test_data_and_symbolicator()
+
+        # Make the in-app frame not symbolicated
+        symbolicator.process_js.return_value["stacktraces"][0]["frames"][0]["data"][
+            "symbolicated"
+        ] = False
+
+        result = process_js_stacktraces(symbolicator, data)
+        self.assertFalse(result["symbolicated_in_app"])
+
+    def test_process_js_stacktraces_with_no_in_app_frames(self):
+        """Test symbolicated_in_app is None when all frames are non-in-app"""
+        data, symbolicator = self._get_test_data_and_symbolicator()
+
+        # Replace all frames with non-in-app frames that will pass _handles_frame
```

Comment:
Can you please make `_get_test_data_and_symbolicator` receive parameters to create the data the way you want rather than updating it within the tests?